src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -222,101 +320,349 @@ export function useSwapInputsController({
+      if (setStale) isQuoteStale.value = 1;
 
-        const updatedSliderPosition = clampJS((amount / Number(assetToSell.value.balance.amount)) * SLIDER_WIDTH, 0, SLIDER_WIDTH);
+      if (inputKey === 'inputAmount') {
+        inputValues.value.inputAmount = amount;
+        await handleInputAmountLogic(amount);
+      } else if (inputKey === 'outputAmount') {
+        inputValues.value.outputAmount = amount;
+        await handleOutputAmountLogic(amount);
+      }
+    } else {
+      const resetValuesToZero = () => {
+        isFetching.value = false;
 
         const updateWorklet = () => {
           'worklet';
+          const keysToReset = ['inputAmount', 'inputNativeValue', 'outputAmount', 'outputNativeValue'];
+          const updatedValues = keysToReset.reduce(
+            (acc, key) => {
+              const castedKey = key as keyof typeof inputValues.value;
+              acc[castedKey] = castedKey === inputKey && preserveAmount ? inputValues.value[castedKey] : 0;
+              return acc;
+            },
+            {} as Partial<typeof inputValues.value>
+          );
           inputValues.modify(values => {
             return {
               ...values,
-              outputAmount,
-              outputNativeValue,
+              ...updatedValues,
             };
           });
-          sliderXPosition.value = withSpring(updatedSliderPosition, snappySpringConfig);
+          sliderXPosition.value = withSpring(0, snappySpringConfig);
           isQuoteStale.value = 0;
         };
 
         runOnUI(updateWorklet)();
-      } else if (inputKey === 'outputAmount') {
-        if (!assetToSell.value || !assetToBuy.value) return;
+      };
 
-        const outputAmount = amount;
-        const inputNativeValue = outputAmount * Number(assetToBuy.value.native.price?.amount) * (1 + SWAP_FEE);
-        const inputAmount = inputNativeValue / Number(assetToSell.value.native.price?.amount);
+      animationFrameId.current = requestAnimationFrame(resetValuesToZero);
+    }
 
-        const updatedSliderPosition = clampJS((inputAmount / Number(assetToSell.value.balance.amount)) * SLIDER_WIDTH, 0, SLIDER_WIDTH);
+    return () => {
+      resetTimers();
+    };
+  }, 400);
 
-        const updateWorklet = () => {
-          'worklet';
-          inputValues.modify(values => {
-            return {
-              ...values,
-              inputAmount,
-              inputNativeValue,
-            };
-          });
-          sliderXPosition.value = withSpring(updatedSliderPosition, snappySpringConfig);
-          isQuoteStale.value = 0;
-        };
+  const fetchAssetPrices = async ({
+    assetToSell,
+    assetToBuy,
+  }: {
+    assetToSell: ParsedSearchAsset | null;
+    assetToBuy: ParsedSearchAsset | null;
+  }) => {
+    const fetchPriceForAsset = async (asset: ParsedSearchAsset | null, assetType: 'assetToSell' | 'assetToBuy') => {
+      if (!asset) return;
+
+      const assetWithPriceData = await fetchExternalToken({
```

Comment:
this looks like the vanilla fetch which wont take advantage of caching/local data

Reply:
added a new ticket: https://linear.app/rainbow/issue/APP-1343/use-react-query-for-fetching-asset-prices-quotes